src/components/Header/web3/NetworkModal/DraggableNetworkButton.tsx

```diff
@@ -0,0 +1,316 @@
+        navigate(
+          {
+            search: stringify(rest),
+          },
+          { replace: true },
+        )
+        onChangedNetwork?.()
+      })
+    } else {
+      changeNetwork(chainId, () => {
+        navigate(
+          {
+            search: '',
+          },
+          { replace: true },
+        )
+        onChangedNetwork?.()
+      })
+    }
+  }
+
+  const variants = {
+    dragging: {
+      boxShadow: '0 5px 10px #00000060',
+      filter: 'brightness(0.8)',
+      scale: 1.05,
+      zIndex: 2,
+    },
+    normal: {
+      boxShadow: '0 0px 0px #00000060',
+      filter: 'brightness(1)',
+      scale: 1,
+      zIndex: 'unset',
+      x: 0,
+      y: 0,
+    },
+  }
+
+  const handleDragStart = () => {
+    animateControls.start('dragging')
+    setIsDragging(true)
+  }
+  const handleDragEnd = (e: any) => {
+    animateControls.start('normal')
+
+    const eventTarget = e.target as HTMLDivElement
+
+    if (!eventTarget) return
+
+    const droppedElPosition = [
+      eventTarget.getBoundingClientRect().left + eventTarget.getBoundingClientRect().width / 2,
+      eventTarget.getBoundingClientRect().top + eventTarget.getBoundingClientRect().height / 2,
+    ]
+
+    const currentEl: HTMLDivElement | undefined = droppableRefs.current?.find((el: HTMLDivElement) => {
+      const { left, top, right, bottom } = el.getBoundingClientRect()
+      return (
+        left < droppedElPosition[0] &&
+        top < droppedElPosition[1] &&
+        right > droppedElPosition[0] &&
+        bottom > droppedElPosition[1]
+      )
+    })
+    if (currentEl) {
+      onDrop?.(currentEl.id)
+    }
+  }
+
+  return (
+    <MouseoverTooltip
+      style={{ zIndex: Z_INDEXS.MODAL + 1 }}
+      key={networkInfo.chainId}
+      text={
+        disabled && !isDragging
+          ? isMaintenance
+            ? t`Chain under maintenance. We will be back as soon as possible`
+            : disabledMsg
+          : ''
+      }
+      width="fit-content"
+      placement="top"
+    >
+      <ListItem
+        drag
+        dragListener={false}
+        dragMomentum={false}
+        dragControls={dragControls}
+        dragConstraints={dragConstraints}
+        dragElastic={0}
+        dragTransition={{ bounceStiffness: 500 }}
+        onDragStart={handleDragStart}
+        onDragEnd={handleDragEnd}
+        onLayoutAnimationStart={() => setIsDragging(true)}
+        onLayoutAnimationComplete={() => setIsDragging(false)}
+        layoutId={networkInfo.chainId.toString() + networkInfo.route}
+        selected={selected}
+        animate={animateControls}
+        transition={{ type: 'spring', damping: 50, stiffness: 1000 }}
+        variants={variants}
+        style={{ boxShadow: '0 0px 0px #00000060' }}
+        onClick={() => !selected && !isDragging && handleChainSelect()}
+        $disabled={disabled}
+      >
+        <img src={icon} alt="Switch Network" style={{ height: '20px', width: '20px' }} />
```

Comment:
```suggestion
        <img src={icon} alt="Switch Network" style={{ maxHeight: '20px', maxWidth: '20px' }} />
```

Reply:
copy from legacy code, I think unnecessary to update